ga4gh/datamodel/rna_quantification.py

```diff
@@ -0,0 +1,390 @@
+                pageSize=pageSize, featureGroupId=featureGroupId)
+        return [FeatureGroup(self, featureGroupEntry) for
+                featureGroupEntry in featureGroupsReturned]
+
+    def getFeatureGroup(self, compoundId):
+        """
+        for now the feature group data is autogenerated by examining the
+        relevant expression data file
+        """
+        rawId = datamodel.FeatureGroupCompoundId.deobfuscate(compoundId)
+        localFeatureGroupId = string.split(rawId, ":")[-1]
+        with self._db as dataSource:
+            featureGroupReturned = dataSource.getFeatureGroupById(
+                localFeatureGroupId)
+
+        if featureGroupReturned is not None:
+            return FeatureGroup(self, featureGroupReturned)
+        else:
+            raise exceptions.FeatureGroupNotFoundException(compoundId)
+
+
+_rnaQuantColumns = [('id', 'TEXT'),
+                    ('annotation_ids', 'TEXT'),
+                    ('description', 'TEXT'),
+                    ('name', 'TEXT'),
+                    ('read_group_id', 'TEXT')]
+
+
+_expressionColumns = [('id', 'TEXT'),
+                      ('name', 'TEXT'),
+                      ('annotation_id', 'TEXT'),
+                      ('expression', 'REAL'),
+                      ('feature_group_id', 'TEXT'),
+                      ('is_normalized', 'BOOLEAN'),
+                      ('raw_read_count', 'REAL'),
+                      ('score', 'REAL'),
+                      ('units', 'TEXT')]
+
+
+class SqliteRNABackend(sqliteBackend.SqliteBackedDataSource):
+    """
+    Defines an interface to a sqlite DB which stores all RNA quantifications
+    in the dataset.
+    """
+    def __init__(self, rnaQuantSqlFile="ga4gh-rnaQuant.db"):
+        super(SqliteRNABackend, self).__init__(rnaQuantSqlFile)
+        self.rnaQuantColumnNames = [f[0] for f in _rnaQuantColumns]
+        self.rnaQuantColumnTypes = [f[1] for f in _rnaQuantColumns]
+        self.expressionColumnNames = [f[0] for f in _expressionColumns]
+        self.expressionColumnTypes = [f[1] for f in _expressionColumns]
+
+    def searchRnaQuantificationsInDb(self, pageToken=0, pageSize=None,
+                                     rnaQuantificationId=None):
+        """
+        :param pageToken: int representing first record to return
+        :param pageSize: int representing number of records to return
+        :param rnaQuantificationId: string restrict search by id
+        :return an array of dictionaries, representing the returned data.
+        """
+        sql = ("SELECT * FROM RNAQUANTIFICATION")
+        sql_args = ()
+        if rnaQuantificationId is not None:
+            sql += " WHERE id = ? "
+            sql_args += (rnaQuantificationId,)
+        sql += sqliteBackend.limitsSql(pageToken, pageSize)
+        query = self._dbconn.execute(sql, sql_args)
+        try:
+            return sqliteBackend.sqliteRows2dicts(query.fetchall())
+        except AttributeError:
```

Comment:
Why return None here, rather than raising an exception directly? This would seem to be the best place to identify where the problem occurs, and it would simplify client code if it didn't have to test for None.